python/pyspark/sql/dataframe.py

```diff
@@ -300,7 +300,7 @@ def take(self, num):
         >>> df.take(2)
         [Row(age=2, name=u'Alice'), Row(age=5, name=u'Bob')]
         """
-        return self.limit(num).collect()
+        return self.coalesce(1).limit(num).collect()
```

Comment:
Can this be made more of a wrapper around the Scala stuff rather than doing its own coalescing?